packages/concerto-core/lib/introspect/declaration.js

```diff
@@ -65,6 +65,7 @@ class Declaration extends Decorated {
             throw new IllegalModelException(`Invalid class name '${this.ast.name}'`, this.modelFile, this.ast.location);
         }
 
+        // should we use resolved name here?
         this.name = this.ast.name;
         this.fqn = ModelUtil.getFullyQualifiedName(this.modelFile.getNamespace(), this.name);
```

Comment:
```suggestion
        // should we use resolved name here?
        this.name = this.ast.name;
        this.fqn = ModelUtil.getFullyQualifiedName(this.modelFile.getNamespace(), this.ast.resolvedName);
```

I think that the value of this.name should stay consistent, but we should fix the value of fqn

Reply:
fqn is here is the local fqn for the cto file, right?

```
namespace child

concept Child {}
```
```
namespace parent

import child.{Child as Kid} 

concept Children extends Kid {}
```

The suggest code would result the fqn to be `parent.Child`, instead of `parent.Children`